### PR TITLE
ci: new collect server scripts

### PR DESCRIPTION
**Pull request by h1dden-da3m0n**
### Description

The changes in this PR are required for the Server + Web CI migration from Azure DevOps to GH Actions.
It adds a new collect server script to replace the old and extracts some its parts wo other GitHub Action Workflows.

A very basic schematic on what triggers what within this repo:
CI-schematic
*edit*: there I a missing trigger link from the server publish-API workflow to the VPS where it uploads the spec to **and** runs `mirrorbits refresh` (see the [server PR](jellyfin/jellyfin#6364) workflow for how)

### Change(s)

* add new collect-server script for OS and fiel bundles
* extracted container image build to GitHub workflow
  * updated Dockerfiles to include OCI labels and their formating
* added editorconfig and updated gitignore

### TODO(s)

* [ ] request new GitHub secrets to be added 
  * [ ] JF_BOT_TOKEN
  * [ ] QUAY_USERNAME - Quay Robot Account Name
  * [ ] QUAY_TOKEN - Quay Robot Account
  * [ ] DOCKERHUB_USERNAME - DockerHub Account Name
  * [ ] DOCKERHUB_TOKEN - DockerHub Token

### Before Merge Checklist:

* [ ] **ALL** TODOs done
* [ ] **AT LEAST 3** 'Core' Members approved
* [ ] merge this PR **FIRST** before `server` and `web`

### Issue(s)

* relates to jellyfin/jellyfin#6364
* relates to jellyfin/jellyfin-web#2897

### CODE REVIEW

**Comment by ferferga** [comment]
JF_BOT_TOKEN, so it doesn't clash with the own GH tokens that GitHub adds to the Actions environments.

File: .github/workflows/publish-meta-images.yaml [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+            echo "STAGE=unstable" >> $GITHUB_ENV
+            echo "FULL_TAG=$(date -u '+%Y%m%d.%H%M')" >> $GITHUB_ENV
+            echo "SHORT_TAG=$(date -u '+%Y%m%d')" >> $GITHUB_ENV
+          fi
+
+          echo "SERVER_VERSION=$(echo ${SERVER_VERSION} | grep -Eo '[0-9]+\.[0-9]+\.[0-9]')" >> $GITHUB_ENV
+          echo "WEB_VERSION=$(echo ${WEB_VERSION} | grep -Eo '[0-9]+\.[0-9]+\.[0-9]')" >> $GITHUB_ENV
+
+
+      - name: Setup sariables
```
Comment (ferferga) [inline .github/workflows/publish-meta-images.yaml:88]:
```suggestion
      - name: Setup variables
```

File: .github/workflows/publish-meta-images.yaml [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+      SERVER_VERSION: ${{ needs.setup-vars.outputs.server-version }}
+      WEB_VERSION: ${{ needs.setup-vars.outputs.web-version }}
+      FULL_TAG: ${{ needs.setup-vars.outputs.full-tag }}
+      SHORT_TAG: ${{ needs.setup-vars.outputs.short-tag }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install qemu dependency
```
Comment (ferferga) [inline .github/workflows/publish-meta-images.yaml:125]: Why not use the same action we use in Vue? It should simplify this setup a lot

Reference: https://github.com/jellyfin/jellyfin-vue/blob/master/.github/workflows/docker-release.yml#L30

**Comment by h1dden-da3m0n** [comment]
closing due to obsolescence (however, work will be continued in jellyfin-sandbox)

**Pull request closed without merging**